pkg/config/update.go

```diff
@@ -176,17 +176,28 @@ func SetProperty(name, value string) Change {
 			}
 
 		case "controller":
+			// controller.default.env.<var>
+			// controller.cluster.name.is.long.<var>
+			if len(path) < 4 {
+				break
+			}
+
+			controllerName := path[1]
+			if len(path) > 4 {
+				controllerName = strings.Join(path[1:len(path)-2], ".")
+			}
+
```

Comment:
Since you don't adjust the `path`, the other references become invalid (i.e. `path[3] == "name"` for `controller.cluster.name.is.long.<var>`). I'd say just update the `path` if it has a length greater than 4:
```go
if l := len(path); l > 4 {
  path[1] = strings.Join(path[1:l-2], ".")
  path = append(path[0:2], path[l-2:]...)
}
```

Something similar needs to be done in `UnsetProperty` below too (since the code is horribly copy and pasted).